src/main/java/com/ibm/crypto/plus/provider/ock/Digest.java

```diff
@@ -343,4 +337,20 @@ protected static boolean validId(long id) {
         //OCKDebug.Msg(debPrefix, methodName,  "Id : " + id);
         return (id != 0L);
     }
+
+    synchronized public Object clone() throws CloneNotSupportedException {
+        Digest copy = (Digest) super.clone();
+        
+        copy.contextFromQueue = false;
+        try {
+            copy.digestId = NativeInterface.DIGEST_copy(
+                this.ockContext.getId(), getId());
+            if (copy.digestId == 0) {
+                throw new CloneNotSupportedException("Copy of native digest context failed.");
+            }
+        } catch (OCKException e) {
+            throw new CloneNotSupportedException(e.getMessage());
```

Comment:
The `CloneNotSupportedException` only accepts a `String` parameter. I could do `e.getStackTrace()` and use a `StringBuilder` to turn the array returned into a single `String` to pass. Would that make sense?

Reply:
OK i see. Yes i think converting the stack trace into a string would be a good idea so that information can be found. Reading through this code it seems like it would be hard for this exception to be thrown but better safe then sorry in this spot.

Reply:
LGTM